frontend-auth-module/projects/sl-auth-lib/src/lib/services/guard/auth.guard.ts

```diff
@@ -0,0 +1,27 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthGuard implements CanActivate {
+
+  constructor(private router: Router) {}
+
+  canActivate(route: any, state: RouterStateSnapshot): boolean | UrlTree | Observable<boolean | UrlTree> | Promise<boolean | UrlTree> {
+    
+    const isAuthenticated = localStorage.getItem('name');
+    if (isAuthenticated) {
+      if (state?.url === '/login' || state?.url === '/signup' || state?.url === '/landing' || state?.url === '/otp' || state?.url === '/reset-password') {
```

Comment:
can you include method of this condition check and see how the whole logic of this guard can be simplified